cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TemplateUtils.java

```diff
@@ -31,6 +43,61 @@ public static Map<String, String> labels(HasMetadataTemplate template)   {
         }
     }
 
+    /**
+     * Add additional volumes for a given pod template and a list of existing volumes.
+     *
+     * @param templatePod     The pod template that contains the additional volumes.
+     * @param existingVolumes The list of existing volumes to which the additional volumes will be added.
+     */
+    public static void addAdditionalVolumes(PodTemplate templatePod, List<Volume> existingVolumes) {
+        if (templatePod.getAdditionalVolumes() != null) {
+            templatePod.getAdditionalVolumes().forEach(volumeConfig -> existingVolumes.add(createVolumeFromConfig(volumeConfig)));
+        }
+    }
+
+    /**
+     * Add additional volume mounts to the given list of volume mounts. Validation is performed to ensure none of the
+     * additional volume mount paths are forbidden.
+     * 
+     * @param volumeMounts           The list of volume mounts to be added to
+     * @param additionalVolumeMounts The list of volume mounts to add
+     * @throws RuntimeException If a forbidden mount path is used.
+     */
+    public static void addAdditionalVolumeMounts(List<VolumeMount> volumeMounts, List<VolumeMount> additionalVolumeMounts) {
+        if (additionalVolumeMounts == null) {
+            return;
+        }
+
+        boolean isForbiddenPath = additionalVolumeMounts.stream().anyMatch(additionalVolume -> !additionalVolume.getMountPath().startsWith(ALLOWED_MOUNT_PATH));
+
+        if (isForbiddenPath) {
```

Comment:
I wonder if we want to disallow mounting any of the Strimzi-internal volumes (already mounted elsewhere in the container). Again, I don't know what kube actually does if you try to do that. And I can't (off the top of my head) see a way this might break things. But it seems weird, and unnecessary for the intended use cases.

Reply:
But this is what we archive with the `/mnt` restriction right? It was chosen in the community call.

Reply:
This is different from limiting the user-defined volumes to `/mnt` mount points. I'm referring to the possibility that a user defines a volume mount (with a `/mnt` mount point) to some internal Strimzi volume (e.g. the volumes used to setup the user's `storage`). 

As I said above, I don't see how this is going to break anything, but it does seem unnecessary, and I thought it worthy of discussion about whether we're happy with this possibility.